src/Common/src/TypeSystem/IL/Stubs/PInvokeMarshallingILEmitter.cs

```diff
@@ -510,11 +462,31 @@ public MethodIL EmitIL()
 
             MethodSignature nativeSig = new MethodSignature(
                 targetMethodSignature.Flags, 0, nativeReturnType, nativeParameterTypes);
-            MethodDesc nativeMethod =
-                new PInvokeTargetNativeMethod(_targetMethod.OwningType, nativeSig, _importMetadata);
+
 
             // Call the native method
-            callsiteSetupCodeStream.Emit(ILOpcode.call, _emitter.NewToken(nativeMethod));
+            ILLocalVariable vNativeFunctionPointer = _emitter.NewLocal(_targetMethod.Context.GetWellKnownType(WellKnownType.IntPtr));
+
+            MetadataType lazyHelperType = _targetMethod.Context.GetHelperType("LazyPInvokeHelpers");
+            if (_importMetadata.Module == "[MRT]" || _importMetadata.Module == "*" || _targetMethod.OwningType == lazyHelperType)
```

Comment:
- `*` condition is fine. 

- `[MRT]` would be ideally replaced with `*` as well. But we can live it for now. Replacing it with `*` would require modifying the .NET Native for UWP.

- `_targetMethod.OwningType == lazyHelperType` should be changed to:
```
if (targeting Windows)
    eager = _importMetadata.Module.StartsWith("api-ms-win-");
else
    eager = _importMetadata.Module.StartsWith("System.Private.");
```

Eventually, we can have command line options or attributes to allow overriding this default policy.